lib/tapioca/helpers/rbi_files_helper.rb

```diff
@@ -96,7 +96,7 @@ def validate_rbi_files(command:, gem_dir:, dsl_dir:, auto_strictness:, gems: [],
       )
       say(" Done", :green)
 
-      errors = Spoom::Sorbet::Errors::Parser.parse_string(res.err)
+      errors = Spoom::Sorbet::Errors::Parser.parse_string(res.err || "")
```

Comment:
`res.err` now has the return type `T.nilable(String)`. I think this is the correct approach because passing an empty string into `parse_string` communicates that there are no errors, which I believe is what a nil response from `res.err` is meant to represent.

Reply:
If this now implies that `res.err` being not `nil` has to mean that we will always have a non-empty list of `errors`, then I would have refactored this so that the condition below checks for `res.err` instead of `error.empty?` and then define `errors` below the `if` block, i.e. once I know that there are definitely errors.

Not sure if the implication as I stated above is true, though.

Reply:
@paracycle That's actually the approach I thought of first, but it turns out that `errors` can be empty even if `res.err` is not nil ([I got this info from the Spoom tests](https://github.com/Shopify/spoom/blob/main/test/spoom/sorbet/errors_test.rb#L15)), so keeping that check seemed safer than checking `res.err` directly.

Reply:
The `nil` value is to indicate that stderr has not been captured, ie. when you pass `capture_err: false`: https://github.com/Shopify/spoom/blob/v1.1.16/lib/spoom/sorbet.rb#L54.

Since in our case we always pass `capture_err: true`: https://github.com/Shopify/tapioca/blob/main/lib/tapioca/helpers/sorbet_helper.rb#L32, we could use `T.must` here.

Reply:
Given the explanation, I think I prefer the `res.err || ""` solution here. Thanks for the additional context.

Reply:
Wouldn't `T.must` make more sense? If the `err` has not been captured that means that something went wrong and we would capture it better with `T.must` than masking the problem by passing an empty string.